superset/views/core.py

```diff
@@ -2608,11 +2601,21 @@ def sql_json(self):
         session.add(query)
         session.flush()
         query_id = query.id
-        session.commit()  # shouldn't be necessary
+        session.commit()
```

Comment:
why'd we remove the comment here? By flushing, i think the commit should happen automatically, so it does seem that this still shouldn't be necessary

Reply:
afaik, flushing actually doesn't actually affect the database, just the session. If something went wrong before a commit, the query creation would be rolled back. I'm actually not sure if `session.flush()` is necessary in this case, but I think `session.commit()` is.